test/test/lock/RaceToLock.java

```diff
@@ -8,81 +8,65 @@
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.concurrent.locks.LockSupport;
-import java.util.Random;
 
 public class RaceToLock {
 
-    private final Lock sharedLock = new ReentrantLock(true);
-    private long sharedWaitTime = 0;
+    // Every SHARED_LOCK_INTERVAL iterations of the loop in runSemiShared, we try to lock a shared lock instead of an uncontended lock
+    private static final long SHARED_LOCK_INTERVAL = 10;
+    private static final long DURATION_MS = 4000;
 
-    private final Random random = new Random();
-    private final Lock[] locks = {
-            new ReentrantLock(true),
-            new ReentrantLock(true),
-            new ReentrantLock(true)
-    };
-    private volatile long randomWaitTime;
+    private static final Lock sharedLock = new ReentrantLock(true);
 
-    private volatile boolean exitRequested;
+    private static volatile boolean exitRequested = false;
```

Comment:
It's a bad practice to initialize `volatile` fields with a default value.
Although it does not make difference here, this is known to have performance impact in other context.

Reply:
Where can I read more about this?

Reply:
578f470ebf4ea08a0a828eaae2fbeddc33e6c9ef

Reply:
@fandreuz Some context in https://bugs.openjdk.org/browse/JDK-6736490 and related issues. In short, semantics of the "volatile" keyword does not allow JIT compiler to easily remove redundant write, and this write is accompanied by the memory barrier which can be expensive on performance critical code paths.